system/wifi.toit

```diff
@@ -0,0 +1,90 @@
+// Copyright (C) 2022 Toitware ApS.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; version
+// 2.1 only.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// The license can be found in the file `LICENSE` in the top level
+// directory of this repository.
+
+import system.services show ServiceDefinition ServiceResource
+
+import esp32
+import .network
+import .modules.wifi
+
+WIFI_CONNECT_TIMEOUT_  ::= Duration --s=10
+WIFI_DHCP_TIMEOUT_     ::= Duration --s=16
+
+class WifiServiceDefinition extends NetworkServiceDefinition:
+  state_/WifiState? := null
+
+  connect client/int -> ServiceResource:
+    return connect client null null
+
+  connect client/int ssid/string? password/string? -> ServiceResource:
```

Comment:
I intend to make this version of connect available through an extended wifi network services. Hopefully, it will just work.